server/src/test/java/org/elasticsearch/cluster/metadata/IndexCreationTaskTests.java

```diff
@@ -138,7 +138,7 @@ public void testApplyDataFromTemplate() throws Exception {
 
         assertThat(result.metaData().index("test").getAliases(), hasKey("alias1"));
         assertThat(result.metaData().index("test").getSettings().get("key1"), equalTo("value1"));
-        assertThat(getMappingsFromResponse(), Matchers.hasKey("mapping1"));
+        assertThat(getMappingsFromResponse(), Matchers.hasKey("type"));
```

Comment:
The changes in this test probably have to do with the question I had above about the (temporary?) introduction of a type argument in the merge method? Just trying to confirm, maybe you can explain briefly.

Reply:
Sort of; the issue is that this test extracts mappings by mocking out a MapperService and then capturing the arguments sent to `merge`; previously this would be a top-level map with the types as keys, and the mappings as values.  Now it's capturing just the mapping, which has a top level of `type`, while previously it was getting an explicit type of `mapping1`.  It's all a bit confusing with the various mocks, unfortunately - I really want to simplify this test further in future and remove as much of the mocking as possible